server/app/auth/AuthIdentityProviderName.java

```diff
@@ -44,6 +46,28 @@ public static AuthIdentityProviderName fromConfig(Config config) {
             + supportedOptions);
   }
 
+  public static AuthIdentityProviderName adminIdentityProviderfromConfig(Config config) {
+    if (!config.hasPath(AUTH_ADMIN_CONFIG_PATH)) {
+      // For a long time, this was the only identity provider for admins
+      return AuthIdentityProviderName.ADFS_ADMIN;
+    }
+    String providerName = config.getString(AUTH_ADMIN_CONFIG_PATH);
+    for (var provider : AuthIdentityProviderName.values()) {
+      if (provider.getValue().equals(providerName)) {
+        return provider;
+      }
+    }
+    String supportedOptions =
+        Arrays.stream(AuthIdentityProviderName.values())
+            .map(AuthIdentityProviderName::getValue)
+            .collect(Collectors.joining(", "));
+    throw new IllegalArgumentException(
+        "Unsupported civiform_admin_idp value: "
+            + providerName
+            + ". Supported values are "
+            + supportedOptions);
```

Comment:
**suggestion (blocking):** could this be replaced with:
```suggestion
    String providerName = config.getString(AUTH_ADMIN_CONFIG_PATH);
    try {
      return AuthIdentityProviderName.valueOf(providerName);
    } catch (IllegalArgumentException e) {
      String supportedOptions =
          Arrays.stream(AuthIdentityProviderName.values())
              .map(AuthIdentityProviderName::getValue)
              .collect(Collectors.joining(", "));
      throw new IllegalArgumentException(
          "Unsupported civiform_admin_idp value: "
              + providerName
              + ". Supported values are "
              + supportedOptions);
    }
```

Reply:
The parameter supplied to `.valueOf()` must be the name (e.g., `"GENERIC_OIDC_ADMIN"`), not the associated value (e.g., `"generic-oidc-admin"`). But it looks so much nicer your way that I created a helper function to mimic the idea.